src/fontra/views/editor/scene-controller.js

```diff
@@ -1462,3 +1533,67 @@ function splitLocation(location, glyphAxes) {
 
   return { fontLocation, glyphLocation };
 }
+
+export function connectTwoDistinctContours(path, firstPointIndex, secondPointIndex) {
+  let selectedPointIndices = [];
+
+  const [firstContourIndex, firstContourPointIndex] =
+    path.getContourAndPointIndex(firstPointIndex);
+  const [secondContourIndex, secondContourPointIndex] =
+    path.getContourAndPointIndex(secondPointIndex);
+
+  let firstContour = path.getUnpackedContour(firstContourIndex);
+  let secondContour = path.getUnpackedContour(secondContourIndex);
+
+  if (!!firstContourPointIndex == !!secondContourPointIndex) {
+    secondContour.points.reverse();
+  }
+
+  if (!firstContourPointIndex) {
+    [firstContour, secondContour] = [secondContour, firstContour];
+  }
+  let selectedContourPointIndex1 = firstContour.points.length - 1;
+  let selectedContourPointIndex2 = selectedContourPointIndex1 + 1;
+  let loneCubicHandle;
+  const lastPointFirstContour = firstContour.points.at(-1);
+  if (lastPointFirstContour.type) {
```

Comment:
The way this is written allows `loneCubicHandle` to be written to twice. Which I think why the following case doesn't work correctly:

- contour 1 ends with a dangling off-curve point
- contour 2 starts with a dangling off-curve point

This case should _not_ result more off-curves being added

This if and the next if should be an if / else if instead, not just to prevent the above bug, but also to make the intention clearer.

Reply:
Thanks for finding this. I hope I understood you right. Here is a short demo video as a reference for the fix:

https://github.com/user-attachments/assets/ab05b8e2-94b0-4541-8f07-cb11a7925b84

I mark this as resolved.